x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java

```diff
@@ -0,0 +1,317 @@
+        event14Source.put("correlation_success1", "C");
+        event14Source.put("correlation_success2", "D");
+
+        Map<String, Object> event15 = new HashMap<>();
+        Map<String, Object> event15Source = new HashMap<>();
+        event15.put("_id", "15");
+        event15.put("_source", event15Source);
+        event15Source.put("@timestamp", "12345678999");
+        event15Source.put("event_type", "failure");
+        event15Source.put("sequence", 44);
+        event15Source.put("correlation_failure1", "C");
+        event15Source.put("correlation_failure2", "D");
+
+        Map<String, Object> sequence = new HashMap<>();
+        List<Map<String, Object>> events = unmodifiableList(asList(event14, event15));
+        List<Map<String, Object>> sequences = singletonList(sequence);
+        Map<String, Object> expectedResponse = singletonMap("hits", singletonMap("sequences", sequences));
+
+        sequence.put("join_keys", asList(44, "C", "D"));
+        sequence.put("events", events);
+
+        final List<String> bulkEntries = getSequencesBulkEntries();
+        StringBuilder builder = new StringBuilder();
+        for (int i = 1; i < 16; i++) {
+            builder.append("{\"index\": {\"_id\":" + i + "}}\n");
+            builder.append(bulkEntries.get(i - 1));
+        }
+        
+        Request request = new Request("POST", index + "/_bulk?refresh");
+        request.setJsonEntity(builder.toString());
+        assertOK(client().performRequest(request));
+
+        return expectedResponse;
+    }
+
+    private void assertResponse(Map<String, Object> expected, Map<String, Object> actual) {
+        if (false == expected.equals(actual)) {
+            NotEqualMessageBuilder message = new NotEqualMessageBuilder();
+            message.compareMaps(actual, expected);
+            fail("Response does not match:\n" + message.toString());
+        }
+    }
+
+    private Map<String, Object> runEql(RestClient client, Request request) throws IOException {
+        Response response = client.performRequest(request);
+        try (InputStream content = response.getEntity().getContent()) {
+            return XContentHelper.convertToMap(JsonXContent.jsonXContent, content, false);
+        }
+    }
+
+    private static String readResource(String location) throws IOException {
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(EqlSearchIT.class.getResourceAsStream(location),
+            StandardCharsets.UTF_8)))
+        {
+            String line = reader.readLine();
+            while (line != null) {
+                if (line.trim().startsWith("//") == false) {
+                    builder.append(line);
+                    builder.append('\n');
+                }
+                line = reader.readLine();
+            }
+            return builder.toString();
+        }
+    }
+
+    private List<String> getSequencesBulkEntries() {
```

Comment:
How about externalizing this and reading each entry line by line.

Reply:
Will do.